Lib/Protocols/IdHL7.pas

```diff
@@ -1249,12 +1465,18 @@ procedure TIdHL7ClientThread.Execute;
             FOwner.FOnDisconnect(FOwner);
             end;
           end;
-        if not Terminated then
+        If TimedOut Then
+          Begin
+          FClient.Disconnect;
+          End
+        Else If Not Terminated Then
           begin
           // we got disconnected. ReconnectDelay applies.
+          LRecTime := Now + ((FOwner.FReconnectDelay / 1000) * {second length} (1 / (24 * 60 * 60)));

```

Comment:
Prefer to use `DateUtils` functions, when available, to manipulate `TDateTime` values

Reply:
changed as requested
https://github.com/IndySockets/Indy/pull/442/commits/7bd8d29be4574608893391b97fcb794d959042b9